yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -1390,24 +1390,49 @@ func IsCurrentUserSuperUser(tconf *TargetConf) (bool, error) {
 	}
 	defer conn.Close(context.Background())
 
-	query := "SELECT rolsuper FROM pg_roles WHERE rolname=current_user"
-	rows, err := conn.Query(context.Background(), query)
-	if err != nil {
-		return false, fmt.Errorf("querying if user is superuser: %w", err)
-	}
-	defer rows.Close()
-
-	var isSuperUser bool
-	if rows.Next() {
-		err = rows.Scan(&isSuperUser)
+	runQueryAndCheckPrivilege := func(query string) (bool, error) {
+		rows, err := conn.Query(context.Background(), query)
 		if err != nil {
-			return false, fmt.Errorf("scanning row for superuser: %w", err)
+			return false, fmt.Errorf("querying if user is superuser: %w", err)
 		}
-	} else {
-		return false, fmt.Errorf("no current user found in pg_roles")
+		defer rows.Close()
+
+		var isProperUser bool
+		if rows.Next() {
+			err = rows.Scan(&isProperUser)
+			if err != nil {
+				return false, fmt.Errorf("scanning row for query: %w", err)
+			}
+		} else {
+			return false, fmt.Errorf("no current user found in pg_roles")
+		}
+		return isProperUser, nil
 	}
 
-	return isSuperUser, nil
+	isSuperUserquery := "SELECT rolsuper FROM pg_roles WHERE rolname=current_user"
```

Comment:
Nit: Can we add a comment above here saying what this query does? Just to differentiate from the lower query